waku/waku_rln_relay/rln_relay.nim

```diff
@@ -300,13 +299,26 @@ proc appendRLNProof*(rlnPeer: WakuRLNRelay,
   msg.proof = proofGenRes.get().encode().buffer
   return true
 
+proc clearNullifierLog(rlnPeer: WakuRlnRelay) =
```

Comment:
just to understand it. what we really want to keep is the `seq[ProofMetadata]` of the epochs that are no older than `MaxEpochGap` right?

I mean, in your case you keep `MaxEpochGap` at max, no matter if they meet this condition. But we would be interested in:

```nim
for epoch in rlnPeer.nullifierLog.keys():
  if (epoch + MaxEpochGap) < now_epoch:
    rlnPeer.nullufierLog.del(epoch)
```

this doesn't require an `OrderedTable` (unsure if ordered tables impact performance).

not a blocker